### PR TITLE
Default to latest behavior when a TFM is not specified

### PR DESCRIPTION
**Pull request by AlexGhiondea**
Instead of requiring that a compatibility switch is set in order to get the
latest behavior for all AppContext switches that are based on TFM, have that
behavior on by default.

### CODE REVIEW

**Comment by weshaggard** [comment]
LGTM

**Comment by JeremyKuhne** [comment]
lgtm

File: src/mscorlib/src/System/AppContext/AppContextDefaultValues.cs [hunk 1]
```diff
@@ -28,26 +28,22 @@ private static void ParseTargetFrameworkName(out string identifier, out string p
         {
             string targetFrameworkMoniker = AppDomain.CurrentDomain.SetupInformation.TargetFrameworkName;
 
-            // If we don't have a TFM then we should default to the 4.0 behavior where all quirks are turned on.
             if (!TryParseFrameworkName(targetFrameworkMoniker, out identifier, out version, out profile))
             {
 #if FEATURE_CORECLR
-                if (CompatibilitySwitches.UseLatestBehaviorWhenTFMNotSpecified)
```
Comment (weshaggard) [inline src/mscorlib/src/System/AppContext/AppContextDefaultValues.cs]: Is the switch UseLatestBehaviorWhenTFMNotSpecified now dead?

**Comment by CIPop** [comment]
LGTM

File: src/mscorlib/src/System/AppContext/AppContextDefaultValues.cs (hunk 1 above)
Comment (AlexGhiondea) [inline src/mscorlib/src/System/AppContext/AppContextDefaultValues.cs]: This is still used to set the legacy [BinaryCompatibility](https://github.com/dotnet/coreclr/blob/775003a4c72f0acc37eab84628fcef541533ba4e/src/mscorlib/src/System/Runtime/Versioning/BinaryCompatibility.cs#L466) switches. I don't think we care about that anymore though.

I will file a separate issue to remove this switch (as it touches quite a few files) and is not really needed right now.

**Pull request merged**